src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -16,136 +18,92 @@ public enum OS
             Windows,
         }
 
-        public static OS BuiltOnOS
+        public static OS HostOS
         {
             get
             {
-#if OS_LINUX
-                return OS.Linux;
-#elif OS_OSX
-                return OS.OSX;
-#elif OS_WINDOWS
-                return OS.Windows;
-#else
-                #error Unknown OS
-#endif
-            }
-        }
-
-        public static OS BuiltForOS
-        {
-            get
-            {
-#if OS_LINUX
-                return OS.Linux;
-#elif OS_OSX
-                return OS.OSX;
-#elif OS_WINDOWS
-                return OS.Windows;
-#else
-                #error Unknown OS
-#endif
-            }
-        }
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    return OS.Linux;
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    return OS.OSX;
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    return OS.Windows;
+                }
 
-        public static OS RunningOnOS
-        {
-            get
-            {
-                // TODO: this should become a real runtime check
-#if OS_LINUX
-                return OS.Linux;
-#elif OS_OSX
-                return OS.OSX;
-#elif OS_WINDOWS
-                return OS.Windows;
-#else
-                #error Unknown OS
-#endif
+                throw new NotImplementedException($"Unsupported OS: {RuntimeInformation.OSDescription}");
             }
         }
 
         public static bool RunningOnWindows
         {
-            get => PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows;
+            get => PlatformUtil.HostOS == PlatformUtil.OS.Windows;
         }
 
         public static bool RunningOnMacOS
         {
-            get => PlatformUtil.RunningOnOS == PlatformUtil.OS.OSX;
+            get => PlatformUtil.HostOS == PlatformUtil.OS.OSX;
         }
 
         public static bool RunningOnLinux
         {
-            get => PlatformUtil.RunningOnOS == PlatformUtil.OS.Linux;
+            get => PlatformUtil.HostOS == PlatformUtil.OS.Linux;
         }
 
         public static bool RunningOnRHEL6
         {
             get
             {
-                // TODO: make this a runtime check
-#if OS_RHEL6
-                return true;
-#else
+                if (!RunningOnLinux || !File.Exists("/etc/redhat-release"))
+                {
+                    return false;
+                }
+
+                try
+                {
+                    string redhatVersion = File.ReadAllText("/etc/redhat-release");
```

Comment:
should we cache this information so we are not reading a file every time this method is called?

Reply:
I wondered about that. `stat`ing the same file multiple times on Linux ought to be cheap, but I can cache the result.